server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1273,8 +1284,14 @@ Pattern getPattern() {
 
         @Override
         public void validate(String value) {
-            if (!pattern.matcher(value).matches()) {
-                throw new IllegalArgumentException("Setting [" + value + "] does not match regex [" + pattern.pattern() + "]");
+            if (isMatching){
+                if (!pattern.matcher(value).matches()) {
+                    throw new IllegalArgumentException("Setting must not contain [" + pattern.pattern() + "]");
```

Comment:
You have the same error for both cases here.  This one should keep the old "does not match" pattern.